### PR TITLE
Add support for travelling bubbles

### LINKED ISSUE #74: Travelling Bubbles
A travelling bubble is attached/pinned to a vehicle while maintaining its relative pose.

### PR DESCRIPTION
**Pull request by JVillella**
closes #74 

This is a first release of the travelling bubbles concept. Travelling bubbles are pinned to a given actor ID and will follow it (spatially) and follow its lifecycle (i.e. if the actor has not been spawned yet or has been terminated the bubble will be inactive).

The new bubble interface,

```python
slow_actor = ...
laner_actor = ...
Bubble(
    zone=PositionalZone(pos=(50, 0), size=(10, 15)),
    margin=5,
    actor=slow_actor,
    pinned_actor_id=Bubble.to_actor_id(laner_actor, mission_group="all"),
    pinned_offset=(-7, 10),
),
```

`to_actor_id(...)` internally uses the new Id interface, which mostly follows the RFC I put up a few months ago. We'll want to apply it to more functions of SMARTS (e.g. egos, boids, etc.)

### CODE REVIEW

File: scenarios/intersections/2lane_t/scenario.py [hunk 1]
```diff
@@ -1,104 +0,0 @@
...
-    scenario,
-    Traffic(
-        flows=[
-            Flow(
-                route=RandomRoute(), rate=3600, actors={TrafficActor(name="car"): 1.0},
-            )
-        ]
-    ),
-    name="random",
-)
```
Comment (davidrusu) [inline scenarios/intersections/2lane_t/scenario.py:104]: Did you mean to remove this?

Comment (JVillella) [reply scenarios/intersections/2lane_t/scenario.py:104]: yea, it's leftover - the map got removed already.

File: scenarios/loop/scenario.py [hunk 2]
```diff
@@ -18,7 +18,35 @@
...
+        social_agent_missions={
+            "all": ([laner_actor], [t.Mission(route=t.RandomRoute())])
+        },
+        bubbles=[
+            t.Bubble(
+                zone=t.PositionalZone(pos=(50, 0), size=(10, 15)),
+                margin=5,
+                actor=slow_actor,
+                pinned_actor_id=t.Bubble.to_actor_id(laner_actor, mission_group="all"),
+                pinned_offset=(-7, 10),
```
Comment (davidrusu) [inline scenarios/loop/scenario.py:47]: I didn't understand what `pinned` meant the first time around, how about `follow_actor_id` or `tracking_actor_id` or `travel_with_actor_id`.

File: smarts/core/bubble_manager.py [hunk 3]
```diff
@@ -108,6 +114,36 @@ def is_admissible(self, vehicle_id: str, other_vehicle_ids: Sequence[int]):
...
+
+            # Bring back to origin
+            geom = translate(geom, xoff=-centroid.x, yoff=-centroid.y)
+            geom = rotate(geom, -self._bubble_heading, "centroid", use_radians=True)
+
+            # Now apply new transformation in "vehicle coordinate space"
+            geom = translate(
+                geom,
+                xoff=self._bubble.pinned_offset[0],
+                yoff=self._bubble.pinned_offset[1],
```
Comment (davidrusu) [inline smarts/core/bubble_manager.py:136]: consider exposing a rotational pinned offset as well?

File: smarts/core/utils/id.py [hunk 4]
```diff
@@ -0,0 +1,49 @@
+"""A helper to ensure consistent naming of IDs within SMARTS Platform."""
+import uuid
+
+
+class Id(str):
```
Comment (davidrusu) [inline smarts/core/utils/id.py:5]: nice

**Review by davidrusu — APPROVED** [review]
Very good.

I'd like to see `pinned` change to something that is more immediately understood.

maybe one of these?
* `follow`
* `track`
* `travel_with`

File: smarts/core/bubble_manager.py [hunk 5]
```diff
@@ -290,8 +344,25 @@ def step_bubble_state(
             exited_airlock_2=exited_airlock_2,
         )
 
+    def _move_travelling_bubbles(self):
+        for bubble in self._active_bubbles():
+            if not bubble.is_travelling:
+                pass
```
Comment (liamchzh) [inline smarts/core/bubble_manager.py:350]: I bet you mean `continue`?

**Review by liamchzh — APPROVED** [review]
Looks great!

**Review by Nickniyihan — APPROVED** [review]

File: smarts/core/bubble_manager.py (hunk 3 above)
Comment (JVillella) [reply smarts/core/bubble_manager.py:136]: Could be valuable!

File: smarts/core/bubble_manager.py (hunk 5 above)
Comment (JVillella) [reply smarts/core/bubble_manager.py:350]: Yes I do :)

File: smarts/sstudio/types.py [hunk 6]
```diff
@@ -567,6 +585,10 @@ class Scenario:
     social_agent_missions: Optional[
         Dict[str, Tuple[Sequence[SocialAgentActor], Sequence[Mission]]]
     ] = None
+    """Every dictionary item {group: (actors, missions)} gets run simultaneously. If
+    actors > 1 and missions = 0 or actors = 1 and missions > 0 we cycle through them
```
Comment (JVillella) [inline smarts/sstudio/types.py:589]: Should be `actors > 1 and missions = 1 or actors = 1 and missions > 1`

**Pull request merged**